### PR TITLE
Returnloan

### PR DESCRIPTION
**Pull request by ej9444**
Hey everyone, this is the completed code, check out when you can.

### CODE REVIEW

File: NewBank/newbank/server/NewBank.java [hunk 1]
```diff
@@ -164,25 +176,133 @@ private String payRequest(CustomerID customerId, String[] splitRequest) {
...
+        ArrayList<Microloan> microloans = database.LoadMicroloans();
+
+        Customer customer = database.LoadCustomerReadWrite(customerId.getKey());
+        Account toAccount = customer.getAccount("Main");
+
+        // Eligibility check - Main account must have existed for at least 3 months
+        LocalDate mainOpeningDate = toAccount.getOpeningDate();
+        long diff = ChronoUnit.MONTHS.between(mainOpeningDate, LocalDate.now());
+        if (diff < 3) {
+            System.out.println("Main account must have existed for at least 3 months.");
```
Comment (Wamacava) [inline NewBank/newbank/server/NewBank.java:235]: We need to remove this print statement

File: NewBank/newbank/server/NewBankServer.java [hunk 2]
```diff
@@ -18,8 +19,9 @@ public void run() {
         System.out.println("New Bank Server listening on " + server.getLocalPort());
         try {
             while (true) {
+                System.out.println("Before accepting the socket");
```
Comment (Wamacava) [inline NewBank/newbank/server/NewBankServer.java]: We need to remove this print statement and fix indentation in this file

**Comment by ej9444** [comment]
Hey guys, new pull request for the completed code.

File: NewBank/newbank/server/NewBank.java [hunk 3]
```diff
@@ -164,25 +177,147 @@ private String payRequest(CustomerID customerId, String[] splitRequest) {
...
+        return "SUCCESS";
+
+    }
+
+    /**
+     * This function should be executed periodically to pay back microloans which are due and
+     * update it to a historical database
+     */
+    public void updateMicroloans() {
+        System.out.println("Updating Microloans...");
```
Comment (acairns-345) [inline NewBank/newbank/server/NewBank.java:284]: Do we want to remove this print statement too?

**Review by acairns-345 — COMMENTED** [review]
Code looks great - I've gone through it all with just one comment :)

File: NewBank/newbank/server/NewBank.java (hunk 3 above)
Comment (Wamacava) [reply NewBank/newbank/server/NewBank.java:284]: I would leave this one, it just goes to the server output. But it's useful for observing when microloans are updated.

**Review by Wamacava — APPROVED** [review]

**Pull request merged**